src/object.d

```diff
@@ -3064,6 +3067,28 @@ Value[] values(T : Value[Key], Value, Key)(T *aa) @property
     assert(s.values.length == 0);
 }
 
+@safe unittest
+{
+    @safe static struct Value
+    {
+        string str;
+        this(this) @safe {}
+    }
+    Value[string] aa;
+    static assert(__traits(compiles, { const _ = aa.values.length; }));
+}
+
+@safe unittest
+{
+    static struct Value
+    {
+        string str;
+        this(this) @system {}
+    }
+    Value[string] aa;
+    static assert(!__traits(compiles, { const _ = aa.values.length; }));
```

Comment:
Why does this static assert fail when

```d
const _ = aa.values.length;
```
errors as

```
`@safe` function `object.__unittest_L3081_C7` cannot call `@system` function `object.values!(Value[string], Value, string).values`
```

Do you have any clue about this, @skoppe?

Reply:
Because the delegate gets inferred as @system

Reply:
Thanks. Adjust tests accordingly.